### DISCUSSION TITLE
Animation Pointerout

### DISCUSSION THREAD

**Discussion by MonsieurTib** [AvaloniaUI/Avalonia#9477] [Q&A] [answered]
I'm trying to animate a control, this control has a 0.1 opacity, when pointerover a 1.0 opacity.
I would like a smooth transition between 1.0 to 0.1 ( not 0.1 to 1.0)
I try this code, but there is no animation when pointer is out

```xml

<Style Selector="Rectangle.red">
            <Setter Property="Height" Value="100"/>
            <Setter Property="Width" Value="100"/>
            <Setter Property="Fill" Value="Red"/>
            <Setter Property="Opacity" Value="0.1"/>
            <Style.Animations>
                <Animation Duration="0:0:0.2"  Easing="QuadraticEaseIn" > 
                    <KeyFrame Cue="0%">
                        <Setter Property="Opacity" Value="1.0"/>
                    </KeyFrame>
                    <KeyFrame Cue="100%">
                        <Setter Property="Opacity" Value="0.1"/>
                    </KeyFrame>
                </Animation>
            </Style.Animations>
        </Style>
        <Style Selector="Rectangle.red:pointerover">
            <Setter Property="Opacity" Value="1"/>
        </Style>

```

**Comment by maxkatz6** [comment, marked as answer]
You need to add this animation to the Style with pointerover, not with the style with "red". As with current code this animation will be executed once when "red" class is set on the control, and nothing will be played on pointerover.

Alternatively, you can use transitions. Transition can be set once on the main style (.red in your case), and ":pointerover" can be kept as it is.
